### DISCUSSION TITLE
Run Application on Obuntu 24.04 with VS 2022 Windows 11

### DISCUSSION THREAD

**Discussion by NewCode82** [AvaloniaUI/Avalonia#15604] [Q&A] [answered]
I installed the Avalonia extension for Visual Studio 2022 on my PC with windows 11 operating system.
I created the Avalonia template project, I released it by publishing with Target Runtime linux-x64 but I don't see any .exe file, I only have the following files:
AvaloniaApplication1.Desktop
AvaloniaApplication1.Desktop.pdb
AvaloniaApplication1.pdb
libHarfBuzzSharp.so
libSkiaSharp.so
How can I run the application on ubuntu...?

**Comment by tbibre** [comment, marked as answer]
Your application is AvaloniaApplication1.Desktop
The .pdb files are debug files, you won't have them if you publish in Release rather than Debug.
the .so files are library used by Avalonia.

A step is needed before you run your application: you need to tell the OS the file is executable, using [chmod](https://en.wikipedia.org/wiki/Chmod) in the directory of the file
`chmod +x AvaloniaApplication1.Desktop`
(You may need a sudo before chmod)

Then you can execute your application by clicking on it or with the command line:
`AvaloniaApplication1.Desktop`
(Again, you may need sudo before)

**Reply by thevortexcloud** [reply]
You should not need sudo to execute the binary. You likely wont need it to give execute permission.

In any case, you can do this all via the GUI. Although the exact instructions will vary by what desktop you pick. KDE (which ships with KUbuntu) is probably the most familiar to someone who has only ever used Windows.

**Reply by NewCode82** [reply]
Thanks...I was able to start the application.
As you have noticed, I am new to the Linux world and this is my first approach

**Reply by thevortexcloud** [reply]
All good. Linux can really confuse people who have only ever used Windows it does a lot of things very differently. Another thing that will likely throw you off is file names/paths are case sensitive. Whereas on Windows they are case insensitive. EG `Hello.txt` and `hello.txt` are different files on Linux but the same file on Windows. I have seen way too many programs that horribly fail due to that.

I would also suggest giving the man pages a read if you get stuck. They are built into most distros and contain pretty much everything you need to know about Linux. Everything from kernel API calls to program documentation has a man page. EG if you run `man hier` in a terminal it will tell you about how the file system is structured under Linux. `man dotnet` will give you lots of information about running the `dotnet` command.

Also to elaborate on the whole `sudo` thing, it essentially means "run this command as administrator".

**Comment by thevortexcloud** [comment]
> but I don't see any .exe file

Linux (generally speaking) does not use exe files. Instead binaries are extensionless. Linux technically does not need extensions at all, instead the OS will check the file header to determine what to do with it (the `file` command will tell you what it thinks a file is). But a lot of software still uses extensions for various reasons ([the list of magic numbers is fairly big, and parsing them can be a pain](https://en.wikipedia.org/wiki/List_of_file_signatures)). As pointed out your binary executable file is `AvaloniaApplication1.Desktop`, however you may run into problems opening this. As most desktop environments will interpret the `.Desktop` extension as a desktop file. To get around that you can either rename the file or open it with a terminal.
